django/middleware/common.py

```diff
@@ -43,8 +46,12 @@ def process_request(self, request):
                 if user_agent_regex.search(request.META['HTTP_USER_AGENT']):
                     raise PermissionDenied('Forbidden user agent')
 
+        # Check for rejected Hosts
+        skip_host_validation = any(pattern.search(request.path.lstrip('/'))
```

Comment:
Why does this logic belong in `CommonMiddleware`? `ALLOWED_HOSTS_EXEMPT` should have no effect in all other `get_host()` calls?